react/features/toolbox/actions.web.js

```diff
@@ -235,13 +235,24 @@ export function setProfileButtonUnclickable(unclickable: boolean): Function {
 export function showDesktopSharingButton(): Function {
     return (dispatch: Dispatch<*>) => {
         const buttonName = 'desktop';
+        const disabledByConfig
+            = APP.conference.desktopSharingDisabledByConfig;
+        const disabledTooltipText
+            = APP.conference.desktopSharingDisabledTooltip;
         const visible
-            = APP.conference.isDesktopSharingEnabled
-                && UIUtil.isButtonEnabled(buttonName);
-
-        dispatch(setToolbarButton(buttonName, {
-            hidden: !visible
-        }));
+            = UIUtil.isButtonEnabled(buttonName)
+                && (APP.conference.isDesktopSharingEnabled
+                        || (disabledByConfig && disabledTooltipText));
+
+        const newState = {
+            enabled: APP.conference.isDesktopSharingEnabled,
```

Comment:
isDesktopSharingEnabled is the final state which takes into account every other aspect (is it supported ? is it disabled on this browser ? was it disabled through config on al browsers ?). I will add a comment.